include/oneapi/dpl/pstl/glue_algorithm_ranges_impl.h

```diff
@@ -28,6 +40,289 @@ namespace oneapi
 {
 namespace dpl
 {
+
+#if _ONEDPL_CPP20_RANGES_PRESENT
+namespace ranges
+{
+
+// [alg.foreach]
+
+namespace __internal
+{
+
+struct __for_each_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _R, typename _Proj = std::identity,
+             std::indirectly_unary_invocable<std::projected<std::ranges::iterator_t<_R>, _Proj>>  _Fun>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> && std::ranges::sized_range<_R>
+
+    std::ranges::borrowed_iterator_t<_R>
+    operator()(_ExecutionPolicy&& __exec, _R&& __r, _Fun __f, _Proj __proj = {}) const
+    {
+        const auto __dispatch_tag = oneapi::dpl::__ranges::__select_backend(__exec);
+        oneapi::dpl::__internal::__ranges::__pattern_for_each(
+            __dispatch_tag, std::forward<_ExecutionPolicy>(__exec), __r, __f, __proj);
+
+        return {std::ranges::begin(__r) + std::ranges::size(__r)};
+    }
+}; //__for_each_fn
+}  //__internal
+
+inline constexpr __internal::__for_each_fn for_each;
+
+// [alg.transform]
+
+namespace __internal
+{
+struct __transform_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _R,
+             std::ranges::random_access_range _OutRange, std::copy_constructible _F, typename _Proj = std::identity>
+    requires oneapi::dpl::is_execution_policy_v<std::remove_cvref_t<_ExecutionPolicy>> && std::ranges::sized_range<_R>
+        && std::ranges::sized_range<_OutRange> && std::indirectly_writable<std::ranges::iterator_t<_OutRange>,
```

Comment:
I suppose, one of the ranges should be allowed to be unsized, similarly to the binary transform. @rarutyun, please correct me if I am wrong here.

Reply:
I guess it is fine, because the binary transform allows only one of the input sequences to be unsized. This is not applicable to the unary one due to having only one input sequence.